src/storage/iceberg_type.cpp

```diff
@@ -0,0 +1,128 @@
+#include "duckdb/common/string_util.hpp"
+#include "utils/iceberg_type.hpp"
+#include "duckdb/common/extra_type_info.hpp"
+#include "rest_catalog/objects/list_type.hpp"
+#include "rest_catalog/objects/map_type.hpp"
+#include "rest_catalog/objects/struct_type.hpp"
+#include "rest_catalog/objects/struct_field.hpp"
+#include "rest_catalog/objects/type.hpp"
+
+namespace duckdb {
+
+string IcebergTypeHelper::LogicalTypeToIcebergType(LogicalType &type) {
+	switch (type.id()) {
+	case LogicalTypeId::TINYINT:
+	case LogicalTypeId::UTINYINT:
+	case LogicalTypeId::SMALLINT:
+	case LogicalTypeId::USMALLINT:
+	case LogicalTypeId::INTEGER:
+		return "int";
+	case LogicalTypeId::BOOLEAN:
+		return "boolean";
+	case LogicalTypeId::VARCHAR:
+		return "string";
+	case LogicalTypeId::DATE:
+		return "date";
+	case LogicalTypeId::UINTEGER:
+	case LogicalTypeId::BIGINT:
+		return "long";
+	case LogicalTypeId::FLOAT:
+		return "float";
+	case LogicalTypeId::DOUBLE:
+		return "double";
+	case LogicalTypeId::DECIMAL: {
+		auto aux = type.AuxInfo()->Cast<DecimalTypeInfo>();
+		return StringUtil::Format("decimal(%d, %d)", aux.width, aux.scale);
+	}
+	case LogicalTypeId::UUID:
+		return "uuid";
+	case LogicalTypeId::BLOB:
+		return "binary";
+	case LogicalTypeId::STRUCT:
+		return "struct";
+	case LogicalTypeId::ARRAY: {
+		// Iceberg doesn't support fixed array lengths
+		return "list";
+	}
+	case LogicalTypeId::LIST:
+		return "list";
+	case LogicalTypeId::TIME:
+		return "time";
+	case LogicalTypeId::TIMESTAMP:
+		return "timestamp";
+	case LogicalTypeId::TIMESTAMP_TZ:
+		return "timestamptz";
+
+	case LogicalTypeId::MAP:
+		return "map";
+	default:
+		throw NotImplementedException("Type %s not supported in Iceberg", LogicalTypeIdToString(type.id()));
+	}
+}
+
+rest_api_objects::Type IcebergTypeHelper::CreateIcebergRestType(LogicalType &type, std::function<idx_t()> get_next_id) {
+	rest_api_objects::Type rest_type;
+
+	switch (type.id()) {
+	case LogicalTypeId::MAP: {
+		rest_type.has_primitive_type = false;
+		rest_type.has_map_type = true;
+		rest_type.map_type = rest_api_objects::MapType();
+		auto aux = type.AuxInfo()->Cast<ListTypeInfo>();
```

Comment:
This is not necessary though

You can use `MapType::KeyType(type);` and ``MapType::ValueType(type);`